docs/design_proposals/sync-auto-support.md

```diff
@@ -0,0 +1,150 @@
+# Title
+
+* Author(s): Appu Goundan (@loosebazooka)
+* Design Shepherd: \<skaffold-core-team-member\>
+* Date: 09/17/2019
+* Status: New
+
+## Background
+
+Currently skaffold does not support `sync` for files that are generated
+during a build. For example when syncing java files to a container, one
+would normally expect `.class` files to be sync'd, but skaffold is
+really only aware of `.java` files in a build.
+
+1. Why is this required?
+  - Having a sync mode that has knowledge of files that need to be built before
```

Comment:
Not your fault, may be we should add it to our design doc proposal, 
Can we list some use cases we want to support. On the top of my mind

1. Users can sync files generated files
2. Users can sync generated files which are bundled inside a tar and inside a container.
3. Users can sync files outside of their current workspace? ( for now users can hook up a bash script which can take input ../../)
4.  Users can sync remote files?

With that in mind, what is the contract between Skaffold and `command` to generate these files.
Would skaffold run the command when it sees a change in the `command.inputs` files? In which case, `inputs` should be non-empty.